KAN-5 feat(user-service): initialize service structure and MongoDB co…

Jisto-m-joy opened the pull request:
…nnection

- Set up Express server with TypeScript
- Connected MongoDB using Mongoose
- Added initial project structure
- Created User model

Jisto-m-joy committed:
KAN-5 feat(user-service): initialize service structure and MongoDB connection

- Set up Express server with TypeScript
- Connected MongoDB using Mongoose
- Added initial project structure
- Created User model

Ajai-3 commented on services/user-service/src/app.ts:
```diff
@@ -1 +1,16 @@
+// import healthRouter from "./routes/health";
+
+dotenv.config();
+connectDB();
```
import mongo db and callign will be on th server.ts not here

Jisto-m-joy replied:
Done

Ajai-3 commented on services/user-service/src/models/user.models.ts:
```diff
@@ -1 +1,23 @@
+  name: {
+    type: String,
+    required: true,
+    trim: true,
```
add phoneNumber field its a important field we are  working on food delivvery app so user verification workign with phone number so it is important

Jisto-m-joy replied:
Done

Ajai-3 commented on services/user-service/src/server.ts:
```diff
@@ -0,0 +1,7 @@
+import app from "./app";
+
+const PORT = process.env.PORT;
```
correct you need to create server with http 
import http from = "http"

const server = http.createServer(app)

server.listen(PORT, () => {
  console.log(`User service running on port ${PORT}`);
}

this is correct

Ajai-3 commented on tsconfig.json:
```diff
@@ -0,0 +1,113 @@
+{
```
delete this file we are workign in mono repo so no need ts config file we already have so no need  to create extra confiig an pakage.json liek files in each service if you have more dobts visit delivery boy service it fine for now.

Jisto-m-joy replied:
Done

Jisto-m-joy committed:
KAN-5 refactor(user-service): restructure server setup and enhance model

- Moved MongoDB connection logic from app.ts to server.ts
- Refactored server setup to use native HTTP server instead of app.listen
- Added phone number field to User model
- Removed redundant tsconfig.json from user-service folder to resolve Git conflict
- Added /health endpoint to verify server status
